### PR TITLE
feat: Add Join Group with Link

### PR DESCRIPTION
**Pull request by tim-vandecasteele**
First of all, thanks for this great project, it was exactly what I was looking for!

I know very little of go. Almost nothing actually, but I wanted to add functionality to join a group with a link. This is exactly what I did in this pull request. 

I have no clue about testing or other pull request hygiene for this project, so let me know if you're interested in this change, and if you are, what should happen to make it ok for consumption.

### CODE REVIEW

File: src/services/group.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+func (service groupService) JoinGroupWithLink(_ context.Context, request domainGroup.JoinGroupWithLinkRequest) (response domainGroup.JoinGroupWithLinkResponse, err error) {
+	whatsapp.MustLogin(service.WaCli)
+
+	jid, err := service.WaCli.JoinGroupWithLink(request.Link)
+	if err != nil {
+		return
+	}
+	fmt.Printf("%+v\n", jid)
```
Comment (aldinokemal) [inline src/services/group.go]: i think we can remove this

**Review by aldinokemal — COMMENTED** [review]
Please change this PR title

`feat: Add Join Group with Link`

**Comment by aldinokemal** [comment]
Would you be able to assist with updating the readme.md file too?

File: src/internal/rest/group.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/gofiber/fiber/v2"
+)
+
+type Group struct {
+	Service domainGroup.IGroupService
+}
+
+func InitRestGroup(app *fiber.App, service domainGroup.IGroupService) Group {
+	rest := Group{Service: service}
+	app.Post("/group/join_with_link", rest.JoinGroupWithLink)
```
Comment (aldinokemal) [inline src/internal/rest/group.go]: I prefer `/group/join-with-link` instead of `/group/join_with_link`

File: src/services/group.go (hunk 1 above)
Comment (tim-vandecasteele) [reply src/services/group.go]: Makes sense, I removed it.

File: src/internal/rest/group.go (hunk 2 above)
Comment (tim-vandecasteele) [reply src/internal/rest/group.go]: Thanks, didn't know what the convention was as there weren't any links composed of multiple words yet, I changed it.

**Comment by tim-vandecasteele** [comment]
Thanks for the feedback, I did the changes and added the URL in the readme, any more changes you want to see?

**Review by aldinokemal — APPROVED** [review]
Thank you for your contribution. I greatly appreciate it, and it looks good to me (LGTM).

**Pull request merged**